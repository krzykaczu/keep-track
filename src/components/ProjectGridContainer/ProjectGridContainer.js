import React from 'react';
import { Grid } from 'styled-css-grid';
import styled from 'styled-components';
import ProjectRowElements from '../ProjectRowElements';
import useHover from '../../hooks/useHover';
import PropTypes from 'prop-types';

const StyledGridContainer = styled.div`
  :hover {
    background: rgba(0, 0, 0, 0.1);
  }
`;

const ProjectGridContainer = ({
  client,
  project,
  dueDate,
  startDate,
  dispatch,
  index,
}) => {
  const [hoverRef, isHovered] = useHover();

  return (
    <StyledGridContainer ref={hoverRef}>
      <Grid columns={12} minRowHeight={'40px'}>
        <ProjectRowElements
          client={client}
          project={project}
          dueDate={dueDate}
          startDate={startDate}
          isHovered={isHovered}
          dispatch={dispatch}
          index={index}
        />
      </Grid>
    </StyledGridContainer>
  );
};

ProjectGridContainer.propTypes = {
  client: PropTypes.string.isRequired,
  project: PropTypes.string.isRequired,
  dueDate: PropTypes.instanceOf(Date).isRequired,
  startDate: PropTypes.instanceOf(Date).isRequired,
  dispatch: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
};

export default ProjectGridContainer;
