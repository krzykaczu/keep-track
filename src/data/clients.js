export const clients = ['Ikea', 'Agata meble', 'Vitra', 'BoConcept'];
