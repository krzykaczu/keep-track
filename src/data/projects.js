export const today = new Date();

export const projects = [
  {
    client: 'Ikea',
    project: '500 pcs of Billy bookstands',
    startDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() - 15
    ),
    dueDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() + 13
    ),
  },
  {
    client: 'Agata meble',
    project: 'Flower pots',
    startDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() - 5
    ),
    dueDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() + 30
    ),
  },
  {
    client: 'BoConcept',
    project: '50 sofas',
    startDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() - 3
    ),
    dueDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() + 50
    ),
  },
  {
    client: 'Vitra',
    project: 'New task chair',
    startDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() - 30
    ),
    dueDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() + 6
    ),
  },
];
