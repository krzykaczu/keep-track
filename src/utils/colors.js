export const colors = {
  alert: {
    gradientStart: '#ff6699',
    gradientEnd: '#b30000',
  },
  warning: {
    gradientStart: '#ffeb99',
    gradientEnd: '#ffcc00',
  },
  normal: {
    gradientStart: '#9fdfbf',
    gradientEnd: '#008000',
  },
};
