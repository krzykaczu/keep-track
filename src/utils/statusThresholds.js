export const statusThresholds = {
  setAlert: 7,
  setWarning: 14,
};
