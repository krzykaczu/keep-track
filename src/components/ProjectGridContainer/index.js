import ProjectGridContainer from './ProjectGridContainer';

export default ProjectGridContainer;
