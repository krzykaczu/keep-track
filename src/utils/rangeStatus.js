import { colors } from './colors';

export const rangeStatus = {
  alert: `linear-gradient(-170deg, ${colors.alert.gradientStart} 0%, ${
    colors.alert.gradientEnd
  } 100%)`,
  warning: `linear-gradient(-170deg, ${colors.warning.gradientStart} 0%, ${
    colors.warning.gradientEnd
  } 100%)`,
  normal: `linear-gradient(-170deg, ${colors.normal.gradientStart} 0%, ${
    colors.normal.gradientEnd
  } 100%)`,
};
