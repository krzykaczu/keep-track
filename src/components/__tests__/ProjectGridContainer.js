import React from 'react';
import ProjectGridContainer from '../ProjectGridContainer';
import 'jest-dom/extend-expect';
import 'jest-styled-components';
import { render, cleanup } from 'react-testing-library';
import { today } from '../../data/projects';
import { addDays } from 'date-fns';
import { colors } from '../../utils/colors';

afterEach(cleanup);

test('ProjectGridContainer renders & dueDate shows up & completeButton is not visible', () => {
  const dueDate = addDays(today, 30);
  const { getByTestId, container } = render(
    <ProjectGridContainer
      key={'project'}
      client={'client'}
      project={'project'}
      dueDate={dueDate}
      startDate={today}
      dispatch={() => {}}
      index={1}
    />
  );
  expect(container).toMatchSnapshot();
  expect(getByTestId('client')).toHaveTextContent('client');
  expect(getByTestId('project')).toHaveTextContent('project');
  expect(getByTestId('range')).toHaveStyle('width: 0%');
  expect(getByTestId('range')).toHaveStyle(
    `background: linear-gradient(-170deg, ${colors.normal.gradientStart} 0%, ${
      colors.normal.gradientEnd
    } 100%)`
  );
  expect(getByTestId('dueDate')).toHaveTextContent(
    dueDate.toLocaleDateString('pl-PL')
  );
  expect(getByTestId('completeButton')).toHaveTextContent('âœ”');
  expect(getByTestId('completeButton')).toHaveStyle('opacity: 0');
});
