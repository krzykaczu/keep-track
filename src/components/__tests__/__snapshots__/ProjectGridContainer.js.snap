// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectGridContainer renders & dueDate shows up & completeButton is not visible 1`] = `
.c2 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c4 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 2;
  grid-row-end: span 1;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c5 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 4;
  grid-row-end: span 1;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c6 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 3;
  grid-row-end: span 1;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c1 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(40px,auto);
  grid-template-columns: repeat(12,1fr);
  grid-gap: 8px;
}

.c8 {
  background: linear-gradient(-170deg,#9fdfbf 0%,#008000 100%);
  width: 0%;
  height: 100%;
  border-radius: inherit;
}

.c7 {
  width: 90%;
  height: 20px;
  border-radius: 10px;
  border: 1px solid #666;
  box-shadow: 0 0 1px 2px rgba(0,0,0,0.1);
}

.c3 {
  all: 'unset';
  opacity: 0;
  cursor: pointer;
  border: none;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  will-change: opacity;
  font-family: inherit;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  background: none;
}

.c0:hover {
  background: rgba(0,0,0,0.1);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        width="1"
      >
        <button
          class="c3"
          data-testid="completeButton"
        >
          <span
            aria-hidden="false"
            aria-label="complete"
            data-testid="emoji"
            role="img"
          >
            âœ”
          </span>
        </button>
      </div>
      <div
        class="c4"
        width="2"
      >
        <span
          data-testid="client"
        >
          client
        </span>
      </div>
      <div
        class="c5"
        width="4"
      >
        <span
          data-testid="project"
        >
          project
        </span>
      </div>
      <div
        class="c4"
        width="2"
      >
        <span
          data-testid="dueDate"
        >
          08/5/2019
        </span>
      </div>
      <div
        class="c6"
        width="3"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
            data-testid="range"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
